### DISCUSSION TITLE
Client-proofing via the Image Editor

### DISCUSSION THREAD

**Discussion by khalwat** [craftcms/cms#8393] [Ideas]
Something that would be extremely useful from a client-proofing POV would be to have a per-asset source config setting that would:

* Have the imager editor pop up as soon as an asset is uploaded (assuming it's an image), allowing the user to choose a focal point, crop the image, etc.
* Have an aspect ratio setting for the cropping tool so they have to crop it to the proper dimensions

If you add the preview as per https://github.com/craftcms/cms/issues/1491 that would make it even more awesome, because the content editor could upload the image, the UI pops up where they get to crop the image, preview how the focal point looks at various responsive sizes, etc., and then save it having a decent assurance of how it will look on the frontend.

**Comment by andris-sevcenko** [comment]
Aspect ratio setting didn't make it into the beta. It is waiting its turn when I take the next pass at image editor features.

**Comment by andris-sevcenko** [comment]
Partially addressed with 57c9757.

**Comment by andris-sevcenko** [comment]
Probably worth making that constraint ratio list customizable. Probably per Volume.

**Comment by davist11** [comment]
We would _love_ to have the constraint ratio list customizable. Even if it's not per volume, and only site-wide, would be really useful.

**Comment by cole007** [comment]
Offering some kind of options for image cropping dimensions in the config file would be extremely useful

**Comment by andris-sevcenko** [comment]
Just resolved #6201 for Craft 3.5 by adding the `imageEditorRatios` setting, which partially resolves this issue as well.
